.github/workflows/tests.yml

```
@@ -0,0 +1,78 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+      - release
+  pull_request:
+    branches:
+      - '*'
+  schedule:
+    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onschedule
+    # > Scheduled workflows run on the latest commit on the default or base branch
+    # i.e. this can only run on master
+    - cron:  '0 11 * * *'
+
+env:
+  zip_filepath: "batch_processing_results.zip"
+
+jobs:
+  batch_processing:
+    name: Test batch_processing.sh
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout SCT
+        uses: actions/checkout@v2
+
+      # install_sct edits ~/.bashrc, but those environment changes don't get passed to subsequent steps in GH Actions.
+      # So, we filter through the .bashrc and pass the values to $GITHUB_ENV and $GITHUB_PATH.
+      # Relevant documentation: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#environment-files
+      - name: Install SCT
+        run: |
+          PIP_PROGRESS_BAR=off ./install_sct -y
+          cat ~/.bashrc | grep "export SCT_DIR" | cut -d " " -f 2 >> $GITHUB_ENV
+          cat ~/.bashrc | grep "export MPLBACKEND" | cut -d " " -f 2 >> $GITHUB_ENV
+          cat ~/.bashrc | grep "export PATH" | grep -o "/.*" | cut -d ':' -f 1 >> $GITHUB_PATH
```

What if you did

```suggestion
      - name: Install SCT
        run: |
          PIP_PROGRESS_BAR=off ./install_sct -y
          . ~/.bashrc
```

Does that work I wonder?

This is what I've been assuming is the closest thing we have to a canonical installation instruction. Using other methods non-canonical installations already gave me some trouble in #3125 because

https://github.com/neuropoly/spinalcordtoolbox/blob/073bac3b0307fb90ebc49c98c09058044fa20bea/.ci.sh#L16

doesn't work right in #3140 installs. I think `. ~/.bashrc` is more reliable, though it might be a toss up.

---

Good thinking. I'll give it a shot. :)

---

No dice: 757830e :(

---

Drat.

Okay maybe this is where this PR and mine *should* be syncing up: we could split `.ci.sh` into:

* `.ci.sh install`
* `.ci.sh test`
* `.ci.sh batch_processing`

The latter two would (using a subroutine instead of copy-pasting between workflow files) both source `~/.bashrc` the same way before running their tests. And then we can use that as a launching point for #2674 like we were talking about at the meeting this week.

---

Since your PR (#3125) is dependent on #3152, and that may take some time, perhaps we could stick with `steps:` here temporarily, then switch to your suggestion as part of a later PR.

---

I've added a note referencing this in 42a6520.